File: web-app/src/lib/features/features/checkpoint.ts
import type { Feature } from '../types';

export const checkpointFeatures = {
  'checkpoint-create': {
    id: 'checkpoint-create',
    title: 'Create Checkpoint',
    description: 'Creates a named checkpoint snapshot for a running session.',
    rpcIds: ['checkpoint:create'],
    componentPaths: [],
    testIds: [
      'TestCreateCheckpoint_MissingSessionID',
      'TestCreateCheckpoint_MissingLabel',
      'TestCreateCheckpoint_SessionNotFound',
      'TestCreateCheckpoint_SessionNotStarted',
    ],
    status: 'stable',
    since: '1.0.0',
  },
  'checkpoint-list': {
    id: 'checkpoint-list',
    title: 'List Checkpoints',
    description: 'Lists all checkpoints associated with a given session.',
    rpcIds: ['checkpoint:list'],
    componentPaths: [],
    testIds: [
      'TestListCheckpoints_MissingSessionID',
      'TestListCheckpoints_SessionNotFound',
      'TestListCheckpoints_ReturnsExistingCheckpoints',
      'TestListCheckpoints_EmptyWhenNoCheckpoints',
    ],
    status: 'stable',
    since: '1.0.0',
  },
} as const satisfies Record<string, Feature>;

File: server/features/checkpoint.go
package features

import "github.com/tstapler/stapler-squad/server/featureregistry"

// CheckpointCreate describes the create-checkpoint RPC.
var CheckpointCreate = featureregistry.Feature{
	ID:          "checkpoint-create",
	Title:       "Create Checkpoint",
	Description: "Creates a named checkpoint snapshot for a running session.",
	RPCIDs:      []string{"checkpoint:create"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

// CheckpointList describes the list-checkpoints RPC.
var CheckpointList = featureregistry.Feature{
	ID:          "checkpoint-list",
	Title:       "List Checkpoints",
	Description: "Lists all checkpoints associated with a given session.",
	RPCIDs:      []string{"checkpoint:list"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

func init() {
	featureregistry.Register(CheckpointCreate)
	featureregistry.Register(CheckpointList)
}
